app/src/ui/app.tsx

```diff
@@ -736,7 +736,7 @@ export class App extends React.Component<IAppProps, IAppState> {
     }
 
     if (repository instanceof CloningRepository || repository.missing) {
```

Comment:
There's a really annoying corner case here that I'll call out because I don't have a good answer right now, but if you're cloning a repository and try to remove it in the application, we don't abort the clone, so you end up in a situation where the clone ends up on disk and not in the app.

I think we need to have the ability to tear down a in-flight clone if the user asks for it, but that's a minor thing in the scope of this PR.